src/base.jl

```diff
@@ -557,6 +579,24 @@ function remove_component!(sys::System, component::T) where {T <: Component}
     clear_units!(component)
 end
 
+"""
+Remove a StaticReserveGroup component from the system by its value.
+"""
+function remove_component!(sys::System, service::T) where {T <: Service}
+    groupservices = collect(get_components(StaticReserveGroup, sys))
+    for groupservice in groupservices
+        if service ∈ get_contributing_services(groupservice)
+            throw(ArgumentError("service $(get_name(_service)) is part of a StaticReserveGroup"))
+            return
+        end
+    end
+    check_component_removal(sys, service)
+    IS.remove_component!(sys.data, service)
+    handle_component_removal!(sys, service)
+    clear_units!(service)
+end
+
```

Comment:
> That raises another question.  What would you like to happen if a user calls `remove_component!(sys, individual_service)` when that service is part of a group?  My inclination is that we should block that and force the user to remove the group before removing a service.  I'll offer to help covering these corner cases.

@daniel-thom Is this a good solution?

Reply:
You can actually make it a little simpler.  The way we set this up is that there is one main `remove_component!` method for a `Component`.  That method calls `check_component_removal` before doing the removal and `handle_component_removal` afterwards.  The default methods for both are no-ops.  The best way for you to solve this problem is to implement your checking logic in a new `check_component_removal` method that takes a Service.

Two minor things:
- `get_components` returns an iterable, so you can remove the call to `collect` and avoid some memory allocations.
- You may want to check if T is a StaticReserveGroup and then return early.

Reply:
Done!

Reply:
Please point it out if I did something not desired.